### PR TITLE
Lesson_1_3

### PR DESCRIPTION
**Pull request by AigulY**
1.  Moving List Item into New "Todo List Item" Component
2. Handling "add todo form" submit
3. Storing "new todo- n react" state

### CODE REVIEW

File: src/AddTodoForm.js [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
-        <div>
-            <form>
-                <label htmlFor="todoTitle">Title</label>
-                <input type="text" id="todoTitle" />
-                <button type="submit">Add</button>
-            </form>
-        </div>
+        <form onSubmit={handleAddTodo}>
+            <label htmlFor="todoTitle">Title</label>
+            <input id="todoTitle" name="title" />
```
Comment (JoHeyJo) [inline src/AddTodoForm.js]: what kind of input is this?

File: src/AddTodoForm.js [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+            <button>Add</button>
```
Comment (JoHeyJo) [inline src/AddTodoForm.js]: what kind of button is this?

Comment (AigulY) [reply src/AddTodoForm.js]: Text input and submit button

File: src/App.js [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
 
 function App() {
-  return (
-    <div>
-      <h1>ToDo List</h1>
-      <AddTodoForm /> {/* Using the AddTodoForm component*/}
-      <TodoList /> {/* Using the TodoList component*/}
-    </div>
-  );
+    const [newTodo, setNewTodo] = React.useState("");
```
Comment (JoHeyJo) [inline src/App.js:6]: Nice job accessing `useState` and providing a default value. Although you won't always get an error for not providing a default value, it's a good habit to get into. It'll be helpful in the long run to provide a default value.

File: src/AddTodoForm.js [hunk 4]
```diff
@@ -1,15 +1,21 @@
-import React from "react";
+import React from 'react';
+
+const AddTodoForm = (props) => {
```
Comment (JoHeyJo) [inline src/AddTodoForm.js]: `props` represents the entire object that's being passed from parent to child. Nice job working with it. Now, what's an advanced technique we could use to directly access each property in `props`? Not only will this give us a clear display of which props are being passed along but it'll also clean up our code when we want to access each prop.

This is a little beyond the scope of the assignment but it's a very common pattern you will encounter. You can merge if you can't solve it. Feel free to ping me for help or for another review if you are able to refactor your code.

File: src/AddTodoForm.js (hunk 2 above)
Comment (JoHeyJo) [reply src/AddTodoForm.js]: go ahead and make the adjustments in the code and resubmit.

File: src/AddTodoForm.js [hunk 5]
```diff
@@ -1,19 +1,19 @@
 import React from 'react';
 
-const AddTodoForm = (props) => {
+const AddTodoForm = ({ onAddTodo }) => {
     const handleAddTodo = (event) => {
         event.preventDefault(); //prevent form from refreshing
         const todoTitle = event.target.title.value;
-        props.onAddTodo(todoTitle);
+        onAddTodo(todoTitle);
```
Comment (JoHeyJo) [inline src/AddTodoForm.js:7]: NIce use of destructuring!

**Review by JoHeyJo — APPROVED** [review]
Looks good! Merge it!

**Pull request merged**